lib/nightingale_web/controllers/api/v1/submission_controller.ex

```diff
@@ -0,0 +1,37 @@
+defmodule NightingaleWeb.API.V1.SubmissionController do
+  use NightingaleWeb, :controller
+
+  alias Nightingale.Search.LocationCheck
+  alias Nightingale.{PositiveLocation, Repo}
+
+  def submit_positive_location(conn, params) do
+    response_map =
+      params
+      |> LocationCheck.validate()
+      |> case do
+        {:ok, _location_check} ->
+          params
+          |> Map.merge(%{"json_blob" => params})
+          |> PositiveLocation.changeset()
+          |> Repo.insert()
+          |> case do
+            {:ok, _} ->
+              %{ok: true}
+            {:error, changeset} ->
+              %{
+                ok: false,
+                errors: get_changeset_errors_as_map(changeset)
+              }
+          end
+
+
+        {:error, changeset} ->
```

Comment:
probably can abstract this out some

Reply:
@jwietelmann should i just add app_version to the location_check validation? I like the way it shows lat/lng missing instead of just location as the PositiveLocations changeset would report